generators/chipyard-helpers/src/main/scala/chipyard-helpers/CodingStyle.scala

```diff
@@ -0,0 +1,53 @@
+class ExampleModule extends Module {
+  // val chisel stuff should be in snake case
+  val temp_wire = WireDefault(...)
+  val my_cool_reg = RegInit(...)
+
+  // scala global defaults should be all caps
+  val DEFAULT_VALUE = 20
+
+  // other scala vars should be in camelcase
+  val myOtherVar = 20
+
+  // scala functions that are associated with the module should be put in an obj
+  // notice how the function is also in camel case since it gives a scala value out
+  def myFoo(intBits: Int, chisel_val: UInt): Int = {
+
+  }
+
+  // notice how the function is also in snake case since it gives a chisel value out
+  def my_bar(intBits: Int, intBits2: Int): UInt = {
```

Comment:
Functions should be camelCase always, only chisel hardware variables get snake_case

Reply:
I originally did what you said... but I do think there is some value in having the functions returning Chisel objects be in snake_case. It makes it clear to end users what is returning a Chisel type or Scala type (strictly based on the snake_case or camelCase of the function name).

Reply:
This contradicts existing helper functions everywhere in chisel, chisel.util, and rocketchip. 

Ex: PopCount, RegNext, OHToUInt, PriorityEncoder

Reply:
I don't want to change the world... so I'm fine not doing this.